tests/rados/test_stretch_n-az_netsplit_scenarios.py

```diff
@@ -0,0 +1,286 @@
+            log.error(
+                "Write ops should be possible, number of objects in the pool has not changed"
+            )
+            raise Exception(
+                f"Pool {pool_name} has {pool_stat['stats']['objects']} objs"
+            )
+
+        log.info(
+            f"Successfully wrote {int(final_objects) - int(init_objects)} on pool {pool_name} in degraded mode\n"
+            f"Proceeding to remove the IPtable rules and recover the cluster from degraded mode"
+        )
+
+        time.sleep(5)
+
+        # Starting to flush IP table rules on all hosts
+        for host in cluster_nodes:
+            log.debug(f"Proceeding to flush iptable rules on host : {host.hostname}")
+            host.exec_command(sudo=True, cmd="iptables -F", long_running=True)
+            log.debug(
+                "Observed that just IP tables flush did not work to bring back the nodes to cluster."
+                f"rebooting the nodes post testing. Rebooting node : {host.hostname}"
+            )
+            host.exec_command(sudo=True, cmd="reboot")
+        log.debug("Sleeping for 30 seconds...")
+        time.sleep(30)
+
+        log.info("Proceeding to do checks post Stretch mode netsplit scenarios")
+
+        if not rados_obj.run_pool_sanity_check():
+            log.error("Checks failed post Site Netsplit scenarios")
+            raise Exception("Post execution checks failed on the Stretch cluster")
+
+    except Exception as err:
+        log.error(f"Hit an exception: {err}. Test failed")
+        log.debug(
+            "Test case expected to fail until bug fix : https://bugzilla.redhat.com/show_bug.cgi?id=2265116"
+        )
+        return 1
+    finally:
+        log.debug("---------------- In Finally Block -------------")
+        # Starting to flush IP table rules on all hosts
+        for host in cluster_nodes:
+            log.debug(f"Proceeding to flush iptable rules on host : {host.hostname}")
+            host.exec_command(sudo=True, cmd="iptables -F", long_running=True)
+            log.debug(
+                "Observed that just IP tables flush did not work to bring back the nodes to cluster."
+                f"rebooting the nodes post testing. Rebooting node : {host.hostname}"
+            )
+            host.exec_command(sudo=True, cmd="reboot")
+
+        rados_obj.rados_pool_cleanup()
+        cmd = f"ceph osd crush rule rm {rule_name}"
+        rados_obj.client.exec_command(cmd=cmd, sudo=True)
+
+        init_time, _ = installer.exec_command(cmd="sudo date '+%Y-%m-%d %H:%M:%S'")
+        log.debug(f"time when test was Ended : {init_time}")
+        if set_debug:
+            log.debug("Removing debug configs on the cluster for mon & osd")
+            rados_obj.run_ceph_command(
+                cmd="ceph config rm mon debug_mon", client_exec=True
+            )
+            rados_obj.run_ceph_command(
+                cmd="ceph config rm osd debug_osd", client_exec=True
+            )
+
```

Comment:
Not  unblocking the  hosts if any exceptions occur during the  after execution of  block_in_out_packets_on_host method in the line#143

Reply:
resolved via discussion